src/hapi/pipelines/utilities/sector.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from typing import Dict
+
+from hdx.scraper.base_scraper import BaseScraper
+from hdx.utilities.dateparse import parse_date
+from sqlalchemy.orm import Session
+
+from hapi.pipelines.database.db_sector import DBSector
+
+logger = logging.getLogger(__name__)
+
+
+class Sector(BaseScraper):
+    def __init__(self, session: Session, datasetinfo: Dict[str, str]):
+        super().__init__(
+            "sector",
+            datasetinfo,
+            dict(),
+        )
+        self._session = session
+        self.data = {}
+
+    def run(self):
+        reader = self.get_reader()
+        headers, iterator = reader.read(self.datasetinfo)
+        for inrow in iterator:
+            code = inrow["#sector +code +acronym"]
+            name = inrow["#sector +name +preferred +i_en"]
+            date = parse_date(inrow["#date +created"])
+            self.data[name] = [code, date]
```

Comment:
Following up on this: at this point it seems that `data` has two different functions. For the custom scrapers, it acts as a way to store all of the data so that it can be put into the database. Then, the way it's being used in the other classes is as a lookup table. Maybe we need to separate these usages?

For example, for the custom scrapers you could just have a `self._scraped_data` list where each row is a dictionary of the scraped data. And that's what can be used in `populate`. 

Then, for the lookup table I would just use a dictionary where the keys are the names and the values are the codes.

Reply:
I agree - it got a little muddled here.  I'll separate the two for the custom scrapers.